File: backend/src/main/java/edu/cit/gaane/caresync/features/consultation/repository/ConsultationRepository.java
package edu.cit.gaane.caresync.features.consultation.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import edu.cit.gaane.caresync.features.consultation.entity.ConsultationEntity;

public interface ConsultationRepository
        extends JpaRepository<ConsultationEntity,Long>{

    List<ConsultationEntity> findByPatientId(Long patientId);

}

File: mobile/app/src/main/java/edu/cit/gaane/caresync/mobile/features/consultation/repository/ConsultationRepository.kt
package edu.cit.gaane.caresync.mobile.features.consultation.repository

import edu.cit.gaane.caresync.mobile.features.authentication.network.RetrofitInstance
import edu.cit.gaane.caresync.mobile.features.consultation.models.ConsultationRequest

class ConsultationRepository {

    suspend fun createConsultation(

        request: ConsultationRequest

    ) = RetrofitInstance.consultationApi.createConsultation(request)

    suspend fun getPatientConsultations(

        patientId: Long

    ) = RetrofitInstance.consultationApi.getPatientConsultations(patientId)

    suspend fun getConsultation(

        consultationId: Long

    ) = RetrofitInstance.consultationApi.getConsultation(consultationId)

    suspend fun updateConsultation(

        consultationId: Long,

        request: ConsultationRequest

    ) = RetrofitInstance.consultationApi.updateConsultation(
        consultationId,
        request
    )

}